R/add_highlight.R

```diff
@@ -0,0 +1,136 @@
+#' @title Highlight a specific strata
+#'
+#' @description S3 method for highlighting a specific strata by lowering the opacity of all other strata.
+#'
+#' @author Tim Treis
+#'
+#' @param gg visR object
+#' @param ... other arguments passed on to the method 
+#' 
+#' @examples
+#' library(visR)
+#'
+#' adtte %>%
+#'   visR::estimate_KM("SEX") %>%
+#'   visR::visr() %>%
+#'   visR::add_CI(alpha = 0.4) %>%
+#'   visR::add_highlight("SEX=M", bg_alpha_multiplier = 0.2)
+#'
+#' @return The input `ggplot2` object with adjusted `alpha` values
+#'
+#' @rdname add_highlight
+#'
+#' @export
+
+add_highlight <- function(gg, ...){
+  UseMethod("add_highlight", gg)
+}
+
+#' @param gg A ggplot created with visR
+#' @param strata Name of the strata to be highlighted as shown in the legend.
+#' @param bg_alpha_multiplier Factor with which the `alpha` values of all but the specified strata will be multiplied.
+#'
+#' @rdname add_highlight
+#' @method add_highlight ggsurvfit
+#' @export
+
+add_highlight.ggsurvfit <- function(gg, 
+                                    strata, 
+                                    bg_alpha_multiplier = 0.2,
+                                    ...) {
+  
+  # Ugly hack to suppress CRAN warning as described here:
+  # https://www.r-bloggers.com/2019/08/no-visible-binding-for-global-variable/#option-two
+  alpha <- colour <- fill <- group <- NULL
+  
+  if (missing(strata)) {
```

Comment:
Can you also check for the structure of strata? It needs to contain an actual strata, part of the data. It needs to contain "=" and it needs to contain one of the values of the strata.
You could remove all blanks within the strata argument before working with the argument to build in some defensiveness eg sex =  M will not be recognized => make it sex=M at the start.

Reply:
Right now I am expecting the user to use exactly what is shown in the legend and present him with the exact options he can pick. I think that is more robust than trying to guess what the user meant, no?
![image](https://user-images.githubusercontent.com/12818242/128488662-f0ed8f1d-98de-40a7-9c32-25cca00d45d9.png)